src/ip.cpp

```diff
@@ -1,49 +1,20 @@
 #include "ip.hpp"
 
+#include <bit>
+
 #include "internal.hpp"
 
 namespace oxen::quic
 {
-    ipv4::ipv4(const std::string& ip)
-    {
-        if (ip.empty() || ip.find(':') != std::string::npos)
-            throw std::invalid_argument{"Cannot parse invalid ipv4 address!"};
-
-        auto rv = inet_pton(AF_INET, ip.c_str(), &addr);
-
-        if (rv < 0)
-            throw std::invalid_argument{"System error (ec:{}): {}"_format(errno, std::system_category().message(errno))};
-
-        if (rv == 0)
-            throw std::invalid_argument{"IPv4 constructor failed to parse input: {}"_format(ip)};
-    }
-
     const std::string ipv4::to_string() const
     {
+        uint32_t net = oxenc::load_host_to_big<uint32_t>(&addr);
```

Comment:
```suggestion
        uint32_t net = oxenc::host_to_big(addr);
```
(No need for the heavier memcpy version of the function when you already have a properly aligned integer.)